### PR TITLE
Fixes fair/shows links throughout the app

### PR DESCRIPTION
**Pull request by ashfurrow**
Discussed in Slack: https://artsy.slack.com/archives/CFNFWD5F0/p1551296889331300

#skip_new_tests

### CODE REVIEW

File: src/lib/Components/Show/ShowArtistsPreview.tsx [hunk 1]
```diff
@@ -24,7 +24,7 @@ export class ShowArtistsPreview extends React.Component<Props> {
     owner_type: Schema.OwnerEntityTypes.Show,
   }))
   handlePress(url: string) {
-    Switchboard.presentNavigationViewController(this.props.Component, url)
+    Switchboard.presentNavigationViewController(this.props.Component || this, url)
```
Comment (ashfurrow) [inline src/lib/Components/Show/ShowArtistsPreview.tsx]: Fixed a nasty bug here.

File: src/lib/Components/Lists/SavedShowItemRow.tsx [hunk 2]
```diff
@@ -18,7 +18,8 @@ const track: Track<Props, {}> = _track
 @track()
 class SavedShowItemRow extends React.Component<Props> {
   handleTap() {
-    Switchboard.presentNavigationViewController(this, this.props.show.href)
+    console.log({ show: this.props.show })
+    Switchboard.presentNavigationViewController(this, this.props.show.href || `show/${this.props.show.id}`)
```
Comment (ashfurrow) [inline src/lib/Components/Lists/SavedShowItemRow.tsx]: This was why we were seeing some shows in the Favourites tab direct back to the Home tab.

Comment (l2succes) [reply src/lib/Components/Lists/SavedShowItemRow.tsx]: Ah... got it

**Comment by l2succes** [comment]
Nice catch @ashfurrow

File: src/lib/Components/Show/ShowArtistsPreview.tsx (hunk 1 above)
Comment (orta) [reply src/lib/Components/Show/ShowArtistsPreview.tsx]: that was the bug?

File: src/lib/Scenes/Home/Components/ForYou/Components/FairsRail.tsx [hunk 3]
```diff
@@ -58,7 +58,7 @@ export class FairsRail extends Component<Props, null> {
       }
 
       const selectionHandler = () => {
-        Switchboard.presentNavigationViewController(this, `${fair.profile.href}?entity=fair`)
+        Switchboard.presentNavigationViewController(this, `${fair.id}?entity=fair`)
```
Comment (orta) [inline src/lib/Scenes/Home/Components/ForYou/Components/FairsRail.tsx]: I just did something like this - doesn't it need to be different depending on whether it has a profile or not? I just had a bug like this in native code

Comment (orta) [reply src/lib/Scenes/Home/Components/ForYou/Components/FairsRail.tsx]: https://github.com/artsy/eigen/pull/2772

**Pull request merged**